src/Autofac/Core/ResolvedParameter.cs

```diff
@@ -34,21 +34,16 @@ namespace Autofac.Core
     /// </summary>
     public class ResolvedParameter : Parameter
     {
-        private readonly Func<ParameterInfo, IComponentContext, bool> _predicate;
-        private readonly Func<ParameterInfo, IComponentContext, object> _valueAccessor;
+        private readonly Func<ParameterInfo, IComponentContext, (bool canResolve, Func<object> valueAccessor)> _resolveParameter;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ResolvedParameter"/> class.
         /// </summary>
-        /// <param name="predicate">A predicate that determines which parameters on a constructor will be supplied by this instance.</param>
-        /// <param name="valueAccessor">A function that supplies the parameter value given the context.</param>
-        public ResolvedParameter(Func<ParameterInfo, IComponentContext, bool> predicate, Func<ParameterInfo, IComponentContext, object> valueAccessor)
+        /// <param name="resolveParameter">A predicate that determines which parameters on a constructor will be supplied
+        ///     by this instance and a function that supplies the parameter value given the context.</param>
+        public ResolvedParameter(Func<ParameterInfo, IComponentContext, (bool canResolve, Func<object> valueAccessor)> resolveParameter)
```

Comment:
I'm sort of wrapped around the axle on this parameter type.

`Func<ParameterInfo, IComponentContext, (bool canResolve, Func<object> valueAccessor)> resolveParameter`

I get that it sort of wraps a `Try` mechanism into a single line, but... man. I'm not a fan, I have to say. This seems super unusable.

The intent of the `predicate` there was to allow a `Try` without having the weird extra thing. Was there no way to just keep that instead?